beacon_node/network/src/sync/network_context/requests/blobs_by_range.rs

```diff
@@ -36,12 +36,30 @@ impl<E: EthSpec> ActiveRequestItems for BlobsByRangeRequestItems<E> {
         if !blob.verify_blob_sidecar_inclusion_proof() {
             return Err(LookupVerifyError::InvalidInclusionProof);
         }
-        if self
-            .items
-            .iter()
-            .any(|existing| existing.slot() == blob.slot() && existing.index == blob.index)
-        {
-            return Err(LookupVerifyError::DuplicatedData(blob.slot(), blob.index));
+
+        // Slots are not consecutive but must be increasing
+        // Blob indices are consecutive and increasing
+        if let Some(prev) = self.items.last() {
+            match blob.slot().cmp(&prev.slot()) {
+                Ordering::Greater => {
+                    if blob.index != 0 {
+                        return Err(LookupVerifyError::NotSorted("non-zero index at slot start"));
+                    }
+                }
+                Ordering::Equal => {
+                    if blob.index == prev.index {
+                        return Err(LookupVerifyError::DuplicatedData(blob.slot(), blob.index));
+                    } else if blob.index != prev.index + 1 {
+                        return Err(LookupVerifyError::NotSorted("non-sequential indices"));
```

Comment:
We could group these two conditions into the "non-sequential" one. But I feel it's good to know if a sender is faulty due to duplicate data or non-sequential